### PR TITLE
[openstack] hierarchical projects: Array of Project instead of Array of Hash.

### PR DESCRIPTION
**Pull request by dhague**
For subtree_as_list and parents_as_list, set subtree and parents attributes as Array of Project instead of Array of Hash.

### CODE REVIEW

**Comment by geemus** [comment]
@Ladas @TerryHowe could you guys help with reviews? Thanks!

**Comment by TerryHowe** [comment]
It is hard to pick out the 'array of projects vs array of hash', but I think the refactoring is great. I haven't had the time lately to go over the details, but generally I like it.

**Comment by dhague** [comment]
@TerryHowe Lines 28 & 31 in projects.rb are the ones that convert the Hash in the response body to a Project object. The changes in identity_v3_spec.rb reflect that the parents/subtree now contains Project objects instead of a Hash.

Hope that helps some,
Darren

**Comment by Ladas** [comment]
I'll try to look at this today.

@dhague I think you might need to remove the last commit from @d063130, or do you have it here on purpose? the commit is from the https://github.com/fog/fog/pull/3650

**Comment by dhague** [comment]
@Ladas: good point - I have removed the last commit from @d063130.

File: lib/fog/openstack/core.rb [hunk 1]
```diff
@@ -69,38 +69,19 @@ module Core
...
-        @openstack_userid = options[:openstack_userid]
-
-        @openstack_domain_name = options[:openstack_domain_name]
-        @openstack_user_domain = options[:openstack_user_domain]
-        @openstack_project_domain  = options[:openstack_project_domain]
-        @openstack_domain_id = options[:openstack_domain_id]
-        @openstack_user_domain_id = options[:openstack_user_domain_id]
-        @openstack_project_domain_id  = options[:openstack_project_domain_id]
+        # Create @openstack_* instance variables from all :openstack_* options
+        options.select{|x|x.to_s.start_with? 'openstack'}.each do |openstack_param, value|
```
Comment (Ladas) [inline lib/fog/openstack/core.rb]: I would maybe prefer explicit %w() list, but I guess it's fine

**Comment by Ladas** [comment]
lgtm :+1:  I can't test keystone v3, I don't have any around right now, but I tested it doesn't break v2 :-)

**Comment by geemus** [comment]
Thanks!

**Pull request merged**